pkg/backup/backuper.go

```diff
@@ -0,0 +1,326 @@
+package backup
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/mattn/go-sqlite3"
+
+	"github.com/pkg/errors"
+)
+
+// BackupFilenamePrefix is the prefix used in every backup file.
+const BackupFilenamePrefix = "tbl_backup"
+
+var (
+	open          = openDatabase
+	newBackupFile = createBackupFile
+)
+
+// Backuper is the process that executes the backup process.
+type Backuper struct {
```

Comment:
The component that was added, responsible for backups.

The way it works:
- It receives in the constructor the database file from which the backup will be taken and a directory that has all the backup files
- It has only one method, the `Backup` that executes the whole logic and in the ends spits the result with some stats of the process